tests/test_pipeline/test_middleware_1f1b.py

```diff
@@ -42,24 +41,44 @@ def run_master(args):
     num_microbatches = args.num_microbatches
     use_checkpoint = args.use_checkpoint
 
-    input_sample = torch.randn((batch_size, dim), device=device)
+    def data_gen(model: str):
+        if model == 'MLP':
+            x = torch.zeros((batch_size, dim))
+            kwargs = dict(x=x)
+            return kwargs
+        elif model == 'DAG_MLP':
+            x = torch.zeros((batch_size, dim))
+            y = torch.zeros((batch_size, dim))
+            kwargs = dict(x=x, y=y)
+            return kwargs
+        else:
+            pass
     
-    def data_gen():
-        x = torch.zeros((batch_size, dim))
-        kwargs = dict(x=x)
-        return kwargs
-    
-    data_kwargs = data_gen()
-    engine = OneFOneBPipelineEngine(partition_fn=partial(partition, data_kwargs),
+    data_kwargs = [
+        data_gen('MLP'),
+        data_gen('DAG_MLP'),
+    ]
+    model = [
+        MLP(dim, stage_num * 3),
+        DAG_MLP(dim, stage_num * 3),
+    ]
+
+    i = 1
```

Comment:
I think using parameterize decorator in colossalai.testing is a more elegant implementation.
For example:
@parameterize('model_cls', [MLP, DAG_MLP])
def test(model_cls):